app/controllers/katello/api/v2/content_view_versions_controller.rb

```diff
@@ -82,9 +82,14 @@ def update
 
     api :PUT, "/content_view_versions/:id/republish_repositories", N_("Forces a republish of the version's repositories' metadata")
     param :id, :number, :desc => N_("Content view version identifier"), :required => true
-    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. (Deprecated)"), deprecated: true
+    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. Dangerous operation when version has repositories with the 'Complete Mirroring' mirroring policy")
     def republish_repositories
-      task = async_task(::Actions::Katello::ContentViewVersion::RepublishRepositories, @content_view_version)
+      mirror_complete_repos = @content_view_version.repositories.joins(:root).where(root: { mirroring_policy: ::Katello::RootRepository::MIRRORING_POLICY_COMPLETE })
+      if mirror_complete_repos.size > 0 && !::Foreman::Cast.to_bool(params[:force])
+        fail HttpErrors::BadRequest, _("Metadata republishing is dangerous on content view versions with repositories with the 'Complete Mirroring' mirroring policy.
+          Update mirroring policies on these repositories #{mirror_complete_repos.pluck(:name)}, or use the 'force' parameter to stop mirroring upstream metadata and regenerate metadata locally.")
```

Comment:
```suggestion
          Change the mirroring policy on these repositories #{mirror_complete_repos.pluck(:name)} and try again, or use the 'force' parameter to regenerate metadata locally. On the next sync, the upstream repository's metadata will overwrite local metadata for 'Complete Mirroring' repositories.")
```

Reply:
`On the next sync, the upstream repository's metadata will overwrite local metadata for 'Complete Mirroring' repositories.`
This is not accurate for CV version repos..Those will not be overwritten by library repo syncs..

Reply:
`New versions of the content view will continue to use upstream metadata for repositories with 'Complete Mirroring'`